### ISSUE TITLE
Crashes when opening a file while "vector as raster" is checked

### ISSUE THREAD

**Issue by Tartarin8** [opentoonz/opentoonz#3294]
As the title says. It does not do that with every project but with my actuel one it's 100% of the time. However, if I uncheck the option, I can open my file and then re-check it, without any problem.

**Device Information:**
 - OS: Windows 8.1
 - OpenToonz Version: 1.4
 - GPU: GTX 660
 - GPU Driver Version: 26.21.14.4587

**Comment by RodneyBaker** [comment]
This seems to be a driver issue but... it's not quite clear what driver you should have.
It almost appears to me that you might have too new a driver and might need to back off a few versions.... not sure.

Where you specify GPU version number (i.e. 26.21.14.4587) that numbering convention doesn't match up directly with the numbering I'm seeing on the Geoforce site where I see things such as "445.87 - WHQL"

In short, I'm not sure but I'd reexamine your driver and see if that helps.

Supposedly the Geoforce Experience utility is suppse to keep drivers up to date. 
 Download at:  https://www.nvidia.com/en-us/geforce/geforce-experience/

**Comment by manongjohn** [comment]
Not sure how you get driver issue from this description.

I do think it's related to #2862. The official cause of that was never found but a temp workaround was made to prevent the crash when loading.

@Tartarin8 Since it appears something in that scene causes the issue 100% of the time, are you able to share the project so someone can try and duplicate it and maybe fix the issue? If it's an issue due to the images you don't wish to share, perhaps replace the drawings with scribbles.

**Comment by RodneyBaker** [comment]
@shun-iwasawa may know something about this issue as he added a comment to code related to avoiding a crash with 'show vector as raster' about 4 months ago:

https://github.com/opentoonz/opentoonz/blame/0dc08facf261210e097186e980ea2ed65dc805c8/toonz/sources/toonzqt/icongenerator.cpp#L890

**Comment by Tartarin8** [comment]
> Where you specify GPU version number (i.e. 26.21.14.4587) that numbering convention doesn't match up directly with the numbering I'm seeing on the Geoforce site where I see things such as "445.87 - WHQL"
I wrote what was in the the Windows device manager, but indeed it is the 445.87 driver (in NVidia
 panel).

Here is the file. I removed all that wasn't causing the bug. It seem those two layers are needed to reproduce: a vector layer and a raster (png) layer. Can you reproduce the bug?

**Comment by RodneyBaker** [comment]
Thanks for the project file @Tartarin8 
I don't get an immediate crash upon opening the scene (or any crash yet) but I am operating with Windows 10 here.

I will continue to investigate but @manongjohn is the one likely to find the problem as I think he has a system running older Windows.

A reason I was interested in your graphics card is that I was trying to gain a baseline from which to explore the problem.  Especially given that you are on Win 8.1.  Seeing the difference in driver numbering just made me curious as to why there was a difference in numbering.

**Comment by shun-iwasawa** [comment]
Unfortunately I couldn't reproduce the crash with my Win10 environment either.

I don't remember match about when I patched #2862 but I guess the problem was due to running some OpenGL-related graphics computation in sub thread.

@Tartarin8

The crash may be occurred when drawing something just after loading the scene. Can you please try the following? :
1. In Preferences, set `Loading > Column Icon` to `On Demand`
1. Move to the `Xsheet` room by clicking the room tab.
1. Undock & Close the Viewer window. Now the room should contain only the Xsheet and the Function Editor panels.
1. Try loading the problematic scene and see if OT crashes or not.

**Comment by Tartarin8** [comment]
I did what you told me to do.  It doesn't crash. I've then also try to normally open the scene from the basic room, it doesn't crash neither! So to crash, "vector as raster" must be on, AND Loading must be "at once".
Hope it helps.

**Comment by manongjohn** [comment]
I am able to reproduce this crash on my Win10 machine.  Here is what I found.

It is crashing while trying to build the image for a column in the first sub-xsheet of the main xsheet.  I edited the sub-xsheet down to the 1st 2 columns, a raster and a vector column.

When there was only a raster or only a vector column in the sub-xsheet, it loaded fine.
When the vector column was in column 1 and raster in column 2 of sub-xsheet, it loaded fine.
When the raster column was in column 1 and vector in column 2 of sub-xsheet, it crashed In Qt5Gui.dll when executing Line 288 in imagebuilders.cpp -> ImageRasterizer::build(int imFlags, void *extData)`

```
284        glPushAttrib(GL_ALL_ATTRIB_BITS);
285        glMatrixMode(GL_MODELVIEW), glPushMatrix();
286        glMatrixMode(GL_PROJECTION), glPushMatrix();
287        {
288          std::unique_ptr<QOpenGLFramebufferObject> fb(  
289              new QOpenGLFramebufferObject(d.lx, d.ly));
290
291          fb->bind();
```

**Comment by RodneyBaker** [comment]
Not sure if this has been resolved or not.
@Al532 Are you still experiencing this problem?

@manongjohn provided steps to reproduce but I haven't been able to confirm.
